### PR TITLE
Personalization

### PR DESCRIPTION
**Pull request by JanPetr**

### CODE REVIEW

File: app/code/community/Algolia/Algoliasearch/Helper/Personalization.php [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+            $this->handleCategories($storeId);
+
+            $this->pushToAlgolia($storeId);
+        }
+    }
+
+    private function handleOrderedProducts($storeId)
+    {
+        $query = 'SELECT sfoi.product_id, sfo.customer_id 
+            FROM sales_flat_order_item sfoi
```
Comment (maxiloc) [inline app/code/community/Algolia/Algoliasearch/Helper/Personalization.php]: I think we need to add the prefix here

File: app/code/community/Algolia/Algoliasearch/Helper/Personalization.php [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+            $this->pushToAlgolia($storeId);
+        }
+    }
+
+    private function handleOrderedProducts($storeId)
+    {
+        $query = 'SELECT sfoi.product_id, sfo.customer_id 
+            FROM sales_flat_order_item sfoi 
+            JOIN sales_flat_order sfo ON sfoi.order_id = sfo.entity_id
```
Comment (maxiloc) [inline app/code/community/Algolia/Algoliasearch/Helper/Personalization.php]: Same here

File: app/code/community/Algolia/Algoliasearch/Helper/Personalization.php [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            ->addIsActiveFilter()
+            ->setStoreId($storeId)
+            ->addFieldToFilter('level', array('gt' => 1));
+
+        $usersWithCategories = array();
+
+        /** @var Mage_Catalog_Model_Category $category */
+        foreach ($categories as $category) {
+            $query = 'SELECT sfo.customer_id, COUNT(sfoi.product_id) as items_purchased FROM 
+                catalog_category_product ccp
```
Comment (maxiloc) [inline app/code/community/Algolia/Algoliasearch/Helper/Personalization.php]: prefix

File: app/code/community/Algolia/Algoliasearch/Helper/Personalization.php [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+            ->setStoreId($storeId)
+            ->addFieldToFilter('level', array('gt' => 1));
+
+        $usersWithCategories = array();
+
+        /** @var Mage_Catalog_Model_Category $category */
+        foreach ($categories as $category) {
+            $query = 'SELECT sfo.customer_id, COUNT(sfoi.product_id) as items_purchased FROM 
+                catalog_category_product ccp 
+                JOIN sales_flat_order_item sfoi ON ccp.product_id = sfoi.product_id
```
Comment (maxiloc) [inline app/code/community/Algolia/Algoliasearch/Helper/Personalization.php]: prefix

File: app/code/community/Algolia/Algoliasearch/Helper/Personalization.php [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+                JOIN sales_flat_order sfo ON sfoi.order_id = sfo.entity_id
+                WHERE ccp.category_id = '.((int) $category->getId()).' AND sfo.state = "complete" AND sfo.customer_id IS NOT NULL AND sfo.store_id = '.((int) $storeId).'
```
Comment (maxiloc) [inline app/code/community/Algolia/Algoliasearch/Helper/Personalization.php]: prefix

File: app/code/community/Algolia/Algoliasearch/Helper/Personalization.php [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+                }
+
+                $this->products[$productId] = $usersIds;
+            }
+        }
+    }
+
+    private function getCategoryProductIds($categoryId)
+    {
+        $query = 'SELECT product_id FROM catalog_category_product WHERE category_id = '.((int) $categoryId);
```
Comment (maxiloc) [inline app/code/community/Algolia/Algoliasearch/Helper/Personalization.php]: prefix

**Comment by JanPetr** [comment]
@maxiloc Thanks for the review and great catch on the prefix thing. I totally forgot about this Magento feature.

> How do we make sure that this can be enabled only by enterprise customer?

Currently we don't. There is no API or website endpoint which would tell us the Algolia plan of the customer. So I just need to make sure, it's well-texted and tell people to enabled Personalization only when they are on Ent plan, otherwise it'll break their search.

**Comment by JanPetr** [comment]
Closing this one as it won't be implemented any time soon

**Pull request closed without merging**